feathr_project/feathr/utils/job_utils.py

```diff
@@ -1,77 +1,187 @@
 import pandas as pd
-import tempfile
 from pandas.errors import EmptyDataError
+from pyspark.sql import DataFrame, SparkSession
+
+from feathr.client import FeathrClient
+from feathr.constants import OUTPUT_FORMAT
 
 
+def get_result_pandas_df(
+    client: FeathrClient,
+    data_format: str = None,
+    res_url: str = None,
+    local_cache_path: str = None,
+) -> pd.DataFrame:
+    """Download the job result dataset from cloud as a Pandas DataFrame.
 
-def get_result_df(client: FeathrClient, format: str = None, res_url: str = None, local_folder: str = None) -> pd.DataFrame:
-    """Download the job result dataset from cloud as a Pandas dataframe to make it easier for the client to read.
+    Args:
+        client: Feathr client
+        data_format: Format to read the downloaded files. Currently support `parquet`, `delta`, `avro`, and `csv`.
+            Default to `avro` if not specified.
+        res_url: Result URL to download files from. Note that this will not block the job so you need to make sure
+            the job is finished and the result URL contains actual data.
+        local_cache_path (optional): Specify the absolute download path. if the user does not provide this,
+            the function will create a temporary directory.
 
-    format: format to read the downloaded files. Currently support `parquet`, `delta`, `avro`, and `csv`. Default to `avro` if not specified.
-    res_url: output URL to download files. Note that this will not block the job so you need to make sure the job is finished and result URL contains actual data.
-    local_folder: optional parameter to specify the absolute download path. if the user does not provide this, function will create a temporary directory and delete it after reading the dataframe.
+    Returns:
+        pandas DataFrame
     """
-    # use a result url if it's provided by the user, otherwise use the one provided by the job 
+    return get_result_df(client, data_format, res_url, local_cache_path)
+
+
+def get_result_spark_df(
+    spark: SparkSession,
+    client: FeathrClient,
+    data_format: str = None,
+    res_url: str = None,
+    local_cache_path: str = None,
+) -> DataFrame:
+    """Download the job result dataset from cloud as a Spark DataFrame.
+
+    Args:
+        spark: Spark session
+        client: Feathr client
+        data_format: Format to read the downloaded files. Currently support `parquet`, `delta`, `avro`, and `csv`.
+            Default to `avro` if not specified.
+        res_url: Result URL to download files from. Note that this will not block the job so you need to make sure
+            the job is finished and the result URL contains actual data.
+        local_cache_path (optional): Specify the absolute download path. if the user does not provide this,
+            the function will create a temporary directory.
+
+    Returns:
+        Spark DataFrame
+    """
+    return get_result_df(client, data_format, res_url, local_cache_path, spark=spark)
```

Comment:
What's the use case of `get_result_spark_df()` and `get_result_pandas_df()` that `get_result_df()` is not enough?

Reply:
`get_result_spark_df` and `get_result_pandas_df` are just an alias to `get_result_df` with more explicit names. I think a function returning two different types of objects depending on arguments are not a good practice and too implicit, but I keep `get_result_df` just because I did not want to remove the previous api which is already in our examples.

Reply:
I see. If you want to expose these two utils functions to endusers, don't forget to add it into `https://github.com/feathr-ai/feathr/blob/main/feathr_project/feathr/__init__.py`.

Reply:
I would like to defer this because the functions I added are not the core functions so that it's okay not able to access via the root module `feathr` and also I don't want to make the root module's name space too heavy.